apps/aecontract/src/aect_sophia.erl

```diff
@@ -61,40 +73,69 @@ parse_options(<<_:8, Rest/binary>>, Acc) ->
     parse_options(Rest, Acc);
 parse_options(<<>>, Acc) -> Acc.
 
+is_legal_serialization_at_height(?SOPHIA_CONTRACT_VSN_1, Height) ->
+    true;
+is_legal_serialization_at_height(?SOPHIA_CONTRACT_VSN_2, Height) ->
+    aec_hard_forks:protocol_effective_at_height(Height) >= ?MINERVA_PROTOCOL_VSN.
+
+serialize(Map) ->
+    serialize(Map, ?SOPHIA_CONTRACT_VSN).
```

Comment:
Do I understand correctly this is only called in debug endpoint for compilation CompileContract now deprecated?

Is caller of that API meant to be able to control serialization? In any case, I would expect swagger.yaml integrated with how CompileContract behaves in interaction with serialization.

Reply:
This has nothing to do with swagger. The serialize/2 function could later potentially be called from the outside if people know what they are doing. Yes, a user should be able to create a serialization of an older version, but Minerva will produce the new version by default once we're there.